docs/typst.md

```diff
@@ -0,0 +1,77 @@
+# Writing high-quality Typst files
+
+No specific code style is mandated, but two spaces of indent and kebab-case for
+variable and function names are recommended.
+
+## Use package specification in imports
+
+When writing a package, it is better to use the full package specification
+when importing your own package, instead of relative imports. In other words,
+it is better to write:
+
+```typ
+#import "@preview/my-package:1.0.0": my-function
+```
+
+than:
+
+```typ
+#import "../lib.typ": my-function
+```
+
+This is especially true for example files. The reasoning here is that
+it should be possible to copy any provided file as is and start working
+from that.
+
+For template files, this is not only a recommendation but a requirement. Users
+should never have to edit a project freshly created from a template to make it
+compile.
+
+This recommendation does not apply to files that are directly part of the package
+however, as this could cause a cyclic import.
+
+## Only exposing specific functions publicly
+
+When writing your package, you will probably create internal functions
+and variables that should not be accessible to the end user. However,
+Typst doesn't provide any keyword to indicate that a given binding should
+be public or private, as most other programming languages do.
+
+Fortunately, there is an easy pattern to select specific items to be public,
+and keep the rest private to your package: the entrypoint of the package
+should not contain any actual code, but simply import the items you want to
+be public from other files.
```

Comment:
```suggestion
Fortunately, there is an easy pattern to select specific items to be public
and keep the rest private to your package: Instead of putting actual code 
into the entrypoint of the package, simply import the items you want to be 
public from other files.
```
I'd like to word this a bit less strongly.